compiler/src/main/kotlin/dev/zacsweers/metro/compiler/ClassIds.kt

```diff
@@ -15,6 +15,7 @@ public class ClassIds(
   customBindsAnnotations: Set<ClassId> = emptySet(),
   customContributesToAnnotations: Set<ClassId> = emptySet(),
   customContributesBindingAnnotations: Set<ClassId> = emptySet(),
+  internal val customContributesIntoSetAnnotations: Set<ClassId> = emptySet(),
```

Comment:
Not adding this annotation to `contributesIntoSetAnnotations` to facilitate additional logic to support `@MapKey`.